hasura/metadata/databases/default/tables/public_api_key.yaml

```diff
@@ -53,19 +70,31 @@ update_permissions:
         - is_active
         - name
       filter:
-        team:
-          users:
-            id:
-              _eq: X-Hasura-User-Id
+        _and:
+          - team:
+              memberships:
+                user_id:
+                  _eq: X-Hasura-User-Id
+          - team_id:
+              _eq: X-Hasura-Team-Id
+          - team:
+              memberships:
```

Comment:
This needs to be nested inside of the other memberships row. Otherwise it's not checking the user is an owner